tool/catalog/generate_widget_catalog.dart

```diff
@@ -42,33 +42,36 @@ Future<void> main(List<String> args) async {
   print('Scanning Dart files...');
   final List<String> libraryFiles = <String>[];
   for (final String file in files) {
-    final SourceKind kind = await session.getSourceKind(file);
-    if (kind == SourceKind.LIBRARY) {
+    final FileResult fileResult = session.getFile(file) as FileResult;
+    if (!fileResult.isPart) {
       libraryFiles.add(file);
     }
   }
   print('  ${libraryFiles.length} dart files');
 
   print("Resolving class 'Widget'...");
 
-  final LibraryElement widgetsLibrary = await session
-      .getLibraryByUri('package:flutter/src/widgets/framework.dart');
+  final LibraryElementResult widgetsLibraryResult = await session
+          .getLibraryByUri('package:flutter/src/widgets/framework.dart')
+      as LibraryElementResult;
+  final LibraryElement widgetsLibrary = widgetsLibraryResult.element;
 
-  final ClassElement widgetClass = widgetsLibrary.getType('Widget');
+  final ClassElement widgetClass =
+      widgetsLibrary.getType('Widget') as ClassElement;
 
   print('Resolving widget subclasses...');
   final List<ClassElement> classes = <ClassElement>[];
   for (final String file in libraryFiles) {
     final ResolvedLibraryResult resolvedLibraryResult =
-        await session.getResolvedLibrary(file);
+        await session.getResolvedLibrary(file) as ResolvedLibraryResult;
 
     final LibraryElement lib = resolvedLibraryResult.element;
     for (final Element element in lib.topLevelElements) {
-      if (element is! ClassElement) {
+      if (element is! ClassElement || element.isMixin) {
```

Comment:
This is the NPE fix, which excludes mixins from the catalog (I assume that's the desired behaviour).